# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/tests/unit/test_cert_chains.js
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 1
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e security/manager/ssl/tests/unit/test_cert_chains.js
--- a/security/manager/ssl/tests/unit/test_cert_chains.js	Mon Oct 21 19:33:02 2019 +0000
+++ b/security/manager/ssl/tests/unit/test_cert_chains.js	Mon Oct 21 19:49:01 2019 +0000
@@ -48,31 +48,6 @@
   );
 }
 
-function test_cert_list_serialization() {
-  let certList = build_cert_chain(["default-ee", "expired-ee"]);
-
-  throws(
-    () => certList.addCert(null),
-    /NS_ERROR_ILLEGAL_VALUE/,
-    "trying to add a null cert to an nsIX509CertList should throw"
-  );
-
-  // Serialize the cert list to a string
-  let serHelper = Cc["@mozilla.org/network/serialization-helper;1"].getService(
-    Ci.nsISerializationHelper
-  );
-  certList.QueryInterface(Ci.nsISerializable);
-  let serialized = serHelper.serializeToString(certList);
-
-  // Deserialize from the string and compare to the original object
-  let deserialized = serHelper.deserializeObject(serialized);
-  deserialized.QueryInterface(Ci.nsIX509CertList);
-  ok(
-    certList.equals(deserialized),
-    "Deserialized cert list should equal the original"
-  );
-}
-
 // We hard-code the following certificates for the pkcs7 export tests so that we
 // don't have to change the test data when the certificates change each year.
 // Luckily these tests don't depend on the certificates being valid, so it's ok
@@ -467,13 +442,13 @@
   deserialized.QueryInterface(Ci.nsITransportSecurityInfo);
 
   equal(
-    deserialized.failedCertChain,
-    null,
-    "failedCertChain for a successful connection should be null"
+    deserialized.failedCertChain.length,
+    0,
+    "failedCertChain for a successful connection should be empty"
   );
   let certChain = build_cert_chain(["default-ee", "test-ca"]);
   ok(
-    certChain.equals(deserialized.succeededCertChain),
+    areCertArraysEqual(certChain, deserialized.succeededCertChain),
     "succeededCertChain should be deserialized correctly"
   );
 }
@@ -546,13 +521,13 @@
   deserialized.QueryInterface(Ci.nsITransportSecurityInfo);
 
   equal(
-    deserialized.succeededCertChain,
-    null,
-    "succeededCertChain should be null"
+    deserialized.succeededCertChain.length,
+    0,
+    "succeededCertChain should be empty"
   );
   let certChain = build_cert_chain(["expired-ee", "test-ca"]);
   ok(
-    certChain.equals(deserialized.failedCertChain),
+    areCertArraysEqual(certChain, deserialized.failedCertChain),
     "failedCertChain should be deserialized correctly"
   );
 }
@@ -572,12 +547,6 @@
     run_next_test();
   });
 
-  // Test serialization of nsIX509CertList
-  add_test(function() {
-    test_cert_list_serialization();
-    run_next_test();
-  });
-
   add_test(function() {
     test_cert_pkcs7_export();
     run_next_test();
@@ -608,8 +577,8 @@
       aTransportSecurityInfo.QueryInterface(Ci.nsITransportSecurityInfo);
       test_security_info_serialization(aTransportSecurityInfo, 0);
       equal(
-        aTransportSecurityInfo.failedCertChain,
-        null,
+        aTransportSecurityInfo.failedCertChain.length,
+        0,
         "failedCertChain for a successful connection should be null"
       );
     }
@@ -634,7 +603,7 @@
       );
       let originalCertChain = build_cert_chain(["expired-ee", "test-ca"]);
       ok(
-        originalCertChain.equals(securityInfo.failedCertChain),
+        areCertArraysEqual(originalCertChain, securityInfo.failedCertChain),
         "failedCertChain should equal the original cert chain for an" +
           " overrideable connection failure"
       );
@@ -657,7 +626,7 @@
       );
       let originalCertChain = build_cert_chain(["unknownissuer"]);
       ok(
-        originalCertChain.equals(securityInfo.failedCertChain),
+        areCertArraysEqual(originalCertChain, securityInfo.failedCertChain),
         "failedCertChain should equal the original cert chain for an" +
           " overrideable connection failure"
       );
@@ -686,7 +655,7 @@
         "test-ca",
       ]);
       ok(
-        originalCertChain.equals(securityInfo.failedCertChain),
+        areCertArraysEqual(originalCertChain, securityInfo.failedCertChain),
         "failedCertChain should equal the original cert chain for a" +
           " non-overrideable connection failure"
       );